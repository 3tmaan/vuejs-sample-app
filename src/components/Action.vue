<template>
  <div class="flex justify-between items-center">
    <p class="pr-2 text-sm">{{action.message}}</p>
    <button class="btn" type="button" @click="undo(index)">Time travel</button>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'Action',
  props: {
    index: {
      type: Number,
      required: true,
    },
  },
  computed: {
    ...mapGetters(["actions"]),
    action() {
      return this.actions[this.index]
    }
  },
  methods: {
    undo(index) {
      const { prevPosts } = this.action;
      this.$store.commit('UNDO_ACTION', {prevPosts, index});
    },
  },
}
</script>

<style scoped>
  .btn {
    @apply py-2 px-4 bg-green rounded outline-none text-sm text-darkgray font-semibold
  }
</style>