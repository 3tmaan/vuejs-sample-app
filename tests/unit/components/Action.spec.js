import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex'
import Action from '@/components/Action.vue';
import mockedActions from '../../../fixtures/actions';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Action.vue', () => {
    let store;
    let mutations;
    let getters;
    let options;

    beforeEach(() => {
        mutations =  {
            UNDO_ACTION: jest.fn(),
        }
        getters = {
            actions: () => mockedActions
        }
        store = new Vuex.Store({
            mutations,
            getters
        })
        options = {
            store,
            localVue,
            propsData: { index: 0 }
        }
    });

    it('renders correctly', () => {
        const wrapper = shallowMount(Action, options);

        expect(wrapper).toMatchSnapshot();
    });

    it('calls store mutations method "UNDO_ACTION" when button is clicked', () => {
        const wrapper = shallowMount(Action, options);
        const button = wrapper.find('button');

        button.trigger('click');
        expect(mutations.UNDO_ACTION).toHaveBeenCalled();
    });
});
