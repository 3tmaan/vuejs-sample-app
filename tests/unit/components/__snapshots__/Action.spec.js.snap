// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action.vue renders correctly 1`] = `
<div class="flex justify-between items-center">
  <p class="pr-2 text-sm">Moved post 1 from index 0 to index 1</p> <button type="button" class="btn">Time travel</button>
</div>
`;
